### ISSUE TITLE
how to record with system sound

### ISSUE THREAD

**Issue by furllmm** [yuka-friends/Windrecorder#187]
and i have problem: my windrecorder consumes battery fast

**Comment by Antonoko** [comment]
Thanks for the feedback! By adding parameters to ffmpeg, it should be easy to record the system sound. So it will be added as an option in the future very soon.
Currently, windrecorder relies heavily on ffmpeg video encoding and decoding, so it will take up a lot of system resources when recording and analyzing. In the future, we will consider refactoring and optimizing it to a mode of continuous screenshot analysis and then delayed compression for video archiving, so as to save performance consumption.

**Comment by furllmm** [comment]
umm if screenshot feature comes is screen recording feature stays in app?
i like screen record feat.

**Comment by Antonoko** [comment]
> umm if screenshot feature comes is screen recording feature stays in app? i like screen record feat. and which config to add? to where ?

yes, currently recording with ffmpeg will take up a lot of system resources. I plan to refactor it to capture screenshot first and then stitch them into videos later to be compatible with the previous implement in the future. However, this method may lose a lot of intermediate images as traditional video recording, so the configuration options of the previous method will also be retained.

**Comment by Antonoko** [comment]
> and i have problem: my windrecorder consumes battery fast

the new version has added a more performance-saving Automatic Flexible Screenshot recording mode and energy-saving options, which should be able to better control power consumption. feel free to update and try it!

**Comment by furllmm** [comment]
how can i reduce ffmpeg ram usage if possible (step by step pls) @Antonoko

**Comment by Antonoko** [comment]
> how can i reduce ffmpeg ram usage if possible (step by step pls) @Antonoko

currently ffmpeg recording uses the gdigrab recording method, which seems requires the CPU to move and encode screen content from the GPU framebuffer, resulting in a large memory overhead. (For my personal use, I prefer to use the flexible screenshot mode, which only requires ffmpeg to compress the video in the end, and does not take up a lot of ram for recording. If the transcoding step still takes up a lot of memory, you can set it to use GPU encoding (Nvidia or AMD) in the advanced encoding settings under compression in the recording settings tab.

It is said that technically, ddagrab/nvenc (nvidia) or dxva2/qsv can be used to replace gdigrab to achieve simultaneous capture and encoding on the GPU. However, due to my limited engineering ability and time and energy, it is still far to achieve this(including multi-monitor and different GPU platforms detective). If you are interested, you can help us study it together. (https://x.com/7id/status/1771091929284894809)

**Comment by furllmm** [comment]
@Antonoko  i not have amd or nvidia gpus 
![image](https://github.com/user-attachments/assets/e6f883ce-c560-4d1a-8240-dec2cd70b501)

**Comment by furllmm** [comment]
can you add intel uhd graphics on that settings

**Comment by Antonoko** [comment]
> can you add intel uhd graphics on that settings

I added `-hwaccel auto` to ffmpeg compress command, hardware acceleration should be automatically called when compressing videos (if available). If you use flexible screenshot mode, you can try it after running install_update.bat. However, video compression itself is still a high-energy task, and it may still take up a lot of resources and cause lag on low-performance hardware.

**Comment by furllmm** [comment]
hmm i detected 2 ffmpeg running after some minute. can u fix?
@Antonoko

**Comment by furllmm** [comment]
i have lenovo ideapad slim 3i laptop 
it has intel uhd graphic only (gpu)
i mean can you add intel uhd graphs to here?
![image](https://github.com/user-attachments/assets/c0d79d91-c495-4ba4-9b31-d1c6eae3696c)

**Comment by Antonoko** [comment]
I'm not too familiar with ffmpeg and whether Intel has unique hardware accelerated instructions and encoding formats, may need to do some research. If anyone knows, please feel free to add information or submit a PR. _(:з」∠)_

**Comment by furllmm** [comment]
i have idea:
join discord server or reddit community about ffmpeg

@Antonoko
